### PR TITLE
MAINT: remove usage of xrange. Closes  #29

### LINKED ISSUE #29: LBFGS not compatible with scipy 1.5.0
The linesearch has been copied from 

the line causing trouble is 
```python
from scipy._lib.six import xrange
```

in the new version they just replaced xrange by range https://github.com/scipy/scipy/blob/4c0fd79391e3b2ec2738bf85bb5dab366dcd12e4/scipy/optimize/linesearch.py#L422

So will I

### PR DESCRIPTION
**Pull request by AntoineSIMTEK**
Now NuMPI should be compatible with scipy 1.5.0

### CODE REVIEW

**Comment by pastewka** [comment]
Something is still broken, can you check @AntoineSIMTEK

**Comment by AntoineSIMTEK** [comment]
I am on it. 

It's a weird error, only with python 3.6. 

```
E               AssertionError: 
E               Not equal to tolerance rtol=1e-07, atol=0
E             
E               Mismatched elements: 1 / 10 (10%)
E               Max absolute difference: 1.44076597e-07
E               Max relative difference: 1.44076597e-07
E                x: array([1., 1., 1., 1., 1., 1., 1., 1., 1., 1.])
E                y: array([1., 1., 1., 1., 1., 1., 1., 1., 1., 1.])
```

**Comment by AntoineSIMTEK** [comment]
Ah I think np.testing.assert_allclose does not displacay the floats with high enough precision

File: tests/test_LBFGS_minimization_problems.py [hunk 1]
```diff
@@ -42,9 +42,9 @@ def test_analytical_min(Objective,n):
     x0 = Objective.startpoint(n)
 
     res = LBFGS(Objective.f, x0, jac=Objective.grad, maxcor=5, gtol=1e-6, maxiter=1000)
-
+    print(np.reshape(res.x, (-1,)))
```
Comment (pastewka) [inline tests/test_LBFGS_minimization_problems.py:45]: Do you want to keep this debug print in there?

**Pull request merged**